### ISSUE TITLE
Confusing error message when email is edited to an existing email

### ISSUE THREAD

**Issue by tanyjnaaman** [tanyjnaaman/pe#3]
Edit command notes indicate that you cannot edit a patron's name, email or id to that of some other patron in LibTask. The natural expectation is that an error message is thrown to indicate which field is duplicated, or at least that those are the fields that have an issue. 

But the error message thrown is that "the person already exists", when something like "the given email/id/name already exists". 

![image.png](https://raw.githubusercontent.com/tanyjnaaman/pe/main/files/5394d083-5a6b-495b-a713-725a95505add.png)

**Comment by nus-pe-script** [comment]
# Team's Response
Thank you for reporting this bug. We have accepted this bug in a separate issue so that you can be credited for this bug report.

Justification for duplication:
As mentioned in the justification for #1470, the way of solving #1470 is to include 3 messages for showing same email, name, and id respectively, in the "Messages" class, and then write a function to print out of of these 3 messages when the patron to be edited or added is considered to be the same as an existing patron based on isSamePatron() function. Once this is done, both #1470 and #300 will be solved simultaneuously.

## The 'Original' Bug

[The team marked this bug as a duplicate of the following bug]

># Duplicate patron
>
>Error message when adding in patrons is not explicit in the reason why its considered a duplicate.
>
>Replicating bug:
>patron add n/Harper Lee s/A0988773M p/91437496 e/harperlee@u.nus.eduuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuu
>patron add n/Harper Lee s/A0988773M p/91437496 e/harperlee@u.nus.eduuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuu
>
>
>![image.png](https://raw.githubusercontent.com/nus-cs2103-AY2122S2/pe/master/files/4f1e419b-6bf6-4eb5-a545-ec99fd9abd21.png)
>
>
>
>
>
>
>
><hr>
><sub>[original: nus-cs2103-AY2122S2/pe-interim#1469] [original labels: severity.Medium type.FeatureFlaw]</sub>
>
>
>

## Their Response to the 'Original' Bug

[This is the team's response to the above 'original' bug]

>Thank you for reporting this issue.
>
>
>![image.png](https://raw.githubusercontent.com/nus-cs2103-AY2122S2/pe/master/files/e126cb4c-9efb-48dc-9dcd-d4930a70be86.png)
>
>As per the module website,  unspecific error messages is at most low severity.
>
>This issue can be solved by adding new error messages for same email, same name and same if respectively in our Message class, and writing a function for printing one of these 3 messages whenever the patron is considered be the same as an existing patron based on isSamePatron() function.


# Items for the Tester to Verify
## :question: Issue duplicate status

Team chose to mark this issue as a duplicate of another issue (as explained in the _**Team's response**_ above)



- [ ] I disagree

**Reason for disagreement:** [replace this with your explanation]

-------------------
## :question: Issue type

Team chose [`type.FeatureFlaw`]
Originally [`type.FunctionalityBug`]

- [ ] I disagree

**Reason for disagreement:** [replace this with your explanation]

-------------------
